var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -61,9 +61,9 @@ class Mpich(AutotoolsPackage, CudaPackage, ROCmPackage):
     variant("wrapperrpath", default=True, description="Enable wrapper rpath")
     variant(
         "pmi",
-        default="pmi",
+        default="default",
         description="""PMI interface.""",
-        values=("pmi", "pmi2", "pmix", "cray"),
+        values=("default", "pmi", "pmi2", "pmix", "cray"),
```

Comment:
Add option `pmi1` for enable only `PMI-v1` protocol. I think `pmi=pmi` should be treated the same as `pmi=default`. In older mpich versions, `--with-pmi=simple` is the default anyway, right?
I think we also should add an option for "slurm", which should be configured as `"--with-pmilib=slurm --with-pmi2={0}".format{spec["slurm-pmi"].prefix`

Reply:
I tried the suggested Slurm configuration among others but I just can't get it to work with the Slurm package. For some reason, it only installs `libpmi2.so`. I think we can probably modify the Slurm package to install both PMI1 and PMI2 and add MPICH support in a separate PR. That or we can default MPICH to use PMI2 with Slurm.